var/spack/repos/builtin/packages/prng/package.py

```diff
@@ -0,0 +1,42 @@
+##############################################################################
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Prng(AutotoolsPackage):
+    """Pseudo-Random Number Generator library."""
+
+    homepage = "http://statmath.wu.ac.at/prng/"
+    url      = "http://statmath.wu.ac.at/prng/prng-3.0.2.tar.gz"
+    list_url = "http://statmath.wu.ac.at/prng"
+
+    version('3.0.2', '80cb0870f2d18618bd2772f9e1dc1a70')
+
+    patch('prng-3.0.2-shared.patch', when="@3.0.2")
+    patch('prng-3.0.2-fix-c99-inline-semantics.patch', when="@3.0.2")
+
+    def autoreconf(self, spec, prefix):
```

Comment:
Why doesn't this happen automatically?

Reply:
It does happen automatically (with a couple of extra steps). Try removing this function and see if it still works. Also, you're missing a few dependencies needed to run `autoreconf`:

- `automake`
- `autoconf`
- `libtool`
- `m4`

Reply:
Well, I tried the automatic way, but there is an already existing `configure` file, so `autoreconf` is never run due to this [check](https://github.com/spack/spack/blob/develop/lib/spack/spack/build_systems/autotools.py#L198).

The problem with the default `configure` is that for some unknown reason it tries to use some specific versions[1] of `automake`, `autoconf`, ... which clash with the ones explicitly installed by Spack. As a result I got this error:

```
==> Installing prng
==> Using cached archive: /spack/var/spack/cache/prng/prng-3.0.2.tar.gz
==> Staging archive: /spack/var/spack/stage/prng-3.0.2-tm66626smkjpcizo6ml7wh6vloa65aiv/prng-3.0.2.tar.gz
==> Created stage in /spack/var/spack/stage/prng-3.0.2-tm66626smkjpcizo6ml7wh6vloa65aiv
==> Applied patch prng-3.0.2-fix-autoconf.patch
==> Applied patch prng-3.0.2-shared.patch
==> Applied patch prng-3.0.2-fix-c99-inline-semantics.patch
==> Building prng [AutotoolsPackage]
==> Executing phase: 'autoreconf'
==> Executing phase: 'configure'
==> Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16'

2 errors found in build log:
     73    param automake-1.7 --gnu Makefile
     74    /spack/var/spack/stage/prng-3.0.2-tm66626smkjpcizo6ml7wh6vloa65aiv/prng-3.0.2/missing: line 47: automake-1.7: command not fo
           und
     75    WARNING: `automake-1.7' is missing on your system.  You should only need it if
     76             you modified `Makefile.am', `acinclude.m4' or `configure.ac'.
     77             You might want to install the `Automake' and `Perl' packages.
     78             Grab them from any GNU archive site.
  >> 79    configure.ac:26: error: possibly undefined macro: AC_PROG_LIBTOOL
     80          If this token and others are legitimate, please use m4_pattern_allow.
     81          See the Autoconf documentation.
  >> 82    make: *** [/spack/var/spack/stage/prng-3.0.2-tm66626smkjpcizo6ml7wh6vloa65aiv/prng-3.0.2/configure] Error 1
```

Reviewing `autotools.py` I've seen there is `force_autoreconf` options that I had missed, which indeed works too, so I will rather use it if you agree. 

I will also define the missing dependencies, it turns out that `which('autoreconf')` was picking the system one so I didn't any error.

-----

[1] Extracted from the original configure file, `${am__api_version}"` is set to `1.7` at the top of the file:
```
# Some tools Automake needs.

ACLOCAL=${ACLOCAL-"${am_missing_run}aclocal-${am__api_version}"}
AUTOCONF=${AUTOCONF-"${am_missing_run}autoconf"}
AUTOMAKE=${AUTOMAKE-"${am_missing_run}automake-${am__api_version}"}
```

Reply:
Yes, `force_autoreconf` is the preferred way.